src/mapper/pkg/kubefinder/kubefinder.go

```diff
@@ -70,6 +71,24 @@ func (k *KubeFinder) initIndexes(ctx context.Context) error {
 		return errors.Wrap(err)
 	}
 
+	err = k.mgr.GetCache().IndexField(ctx, &corev1.Pod{}, podIPIncludingHostNetworkIndexField, func(object client.Object) []string {
+		res := make([]string, 0)
+		pod := object.(*corev1.Pod)
+
+		// TODO: SHOULD I REMOVE IT??
```

Comment:
This is here to say "Should we do some sort of pod IP caching"

Reply:
rather than "TODO", document what features might be missing / broken by the current design choice.